numpy/_core/src/multiarray/array_coercion.c

```diff
@@ -471,7 +472,7 @@ npy_cast_raw_scalar_item(
  * @return 0 on success -1 on failure.
  */
 NPY_NO_EXPORT int
-PyArray_Pack(PyArray_Descr *descr, char *item, PyObject *value)
+PyArray_Pack(PyArray_Descr *descr, void *item, PyObject *value)
```

Comment:
I have a bit of a habit of using `char` since we do pointer arithmetic with strides so much.  But I think here `void *` is the normal thing clearly.